source/include/core_mqtt_agent_default_logging.h

```diff
@@ -0,0 +1,129 @@
+/*
+ * coreMQTT-Agent v1.1.0
+ * Copyright (C) 2022 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
```

Comment:
Should this include core_mqtt_agent_config_default if a user chooses to implement their own logging? i can foresee a file that includes this first then core_mqtt_agent_config_default then core_mqtt_agent_config/core_mqtt_agent and that causing warnings that logging has been redefined. perhaps these logging macros should just be moved to the default config file instead of having two files? or if the plan is to separate config from logging then probably have an option for it similar to the default config file and customers then implement core_mqtt_agent_config.h if DO_NOT_USE_CUSTOM_CONFIG is set and core_mqtt_agent_logging.h if DO_NOT_USE_LOGGING is set?

Reply:
I did not understand properly. The logging config and the other config is completely separate. We should not mix those.
The logging configuration is supposed to be in `core_mqtt_agent_config.h` file.
The user provided configuration will override the definitions in both the lib-config and logging-config files.

So, the user only has to include one config file.

As for being redefined - it should not happen as the logging macros are guarded by ifndef

Reply:
I guess I was a bit confused as to why this file is needed in the first place - to me it feels like a potential trap because if a file includes this file before actual logging definitions you will get re-definition warnings (i.e. including this file then core_mqtt_agent.h). not a huge gripe. i do think that this should be included in core_mqtt_agent.h after core_mqtt_agent_defaults.h instead of included in core_mqtt_agent.c as it is. I'd also rec just having these default logging configs in core_mqtt_agent_config_defaults.h since that would be consistent with https://github.com/FreeRTOS/coreHTTP/blob/main/source/include/core_http_config_defaults.h.

Reply:
Thank you @jasonpcarroll for approving the PR. I am putting the excerpt of our conversation here for the next reader.

- The default headers are not supposed to be included by any application(s).
- Moving the includes of default header files to core_mqtt_agent.h will essentially lead to leaking of logging configuration to the other files using the Agent (like OTA).
- The core_mqtt_agent.h still has included a config file but that cannot be fixed without modifying the API as the MQTTAgent struct declares a configurable array to hold information about unacknowledged packets.

While this doesn't fix the leaking of logging macros on its own, it is a step in the right direction - which we would continue to move towards.